modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java

```diff
@@ -488,7 +492,7 @@ private boolean removeCheckpoint(CheckpointEntry checkpoint) {
                 }
             }
 
-            if ((F.isEmpty(modifiedPartsCounter) && F.isEmpty(historyPointerCandidate)) || ptr.compareTo(latestReservedPointer) == 0)
+            if (F.isEmpty(modifiedPartsCounter))
```

Comment:
It is correct because partsCounter limited by the preloading reservation. This collection received from supplier and is confirmed by coordinator logic.
If we trust a supplier, we can trust every of the demand messages from it.

Reply:
Ok, this should not be relevant if searchEarliestWalPointer is removed from historicalIterator initialization.